x-pack/plugins/lens/public/xy_visualization/xy_expression.tsx

```diff
@@ -198,25 +208,25 @@ export function XYChart({
   }
 
   // use formatting hint of first x axis column to format ticks
-  const xAxisColumn = Object.values(data.tables)[0].columns.find(
-    ({ id }) => id === layers[0].xAccessor
+  const xAxisColumn = data.tables[filteredLayers[0].layerId].columns.find(
+    ({ id }) => id === filteredLayers[0].xAccessor
   );
   const xAxisFormatter = formatFactory(xAxisColumn && xAxisColumn.formatHint);
 
   // use default number formatter for y axis and use formatting hint if there is just a single y column
   let yAxisFormatter = formatFactory({ id: 'number' });
-  if (layers.length === 1 && layers[0].accessors.length === 1) {
+  if (filteredLayers.length === 1 && filteredLayers[0].accessors.length === 1) {
     const firstYAxisColumn = Object.values(data.tables)[0].columns.find(
-      ({ id }) => id === layers[0].accessors[0]
+      ({ id }) => id === filteredLayers[0].accessors[0]
     );
     if (firstYAxisColumn && firstYAxisColumn.formatHint) {
       yAxisFormatter = formatFactory(firstYAxisColumn.formatHint);
     }
   }
 
   const chartHasMoreThanOneSeries =
-    layers.length > 1 || data.tables[layers[0].layerId].columns.length > 2;
-  const shouldRotate = isHorizontalChart(layers);
+    filteredLayers.length > 1 || filteredLayers.some(layer => layer.accessors.length > 2);
```

Comment:
this change prevents the legend to be rendered. I think it should be corrected to 
```suggestion
    filteredLayers.length > 1 || filteredLayers.some(layer => layer.accessors.length > 1);
```
- one of the column is xAccessor and the rest will be accessors. If we have >1, we need a legend. What do you think @wylieconlon ? That also would fix the tests (they break due to the missing legend)